code/game/objects/structures/tables_racks.dm

```diff
@@ -208,6 +214,35 @@
 	material_flags = MATERIAL_ADD_PREFIX | MATERIAL_COLOR
 	buildstack = null //No buildstack, so generate from mat datums
 
+///Table on wheels
+/obj/structure/table/rolling
+	name = "Rolling table"
+	anchored = FALSE
+	smooth = SMOOTH_FALSE
+	canSmoothWith = list()
+	icon = 'icons/obj/smooth_structures/rollingtable.dmi'
+	icon_state = "rollingtable"
+	var/list/attached_items = list()
+
+/obj/structure/table/rolling/AfterPutItemOnTable(obj/item/I, mob/living/user)
+	. = ..()
+	attached_items += I
+	RegisterSignal(I, COMSIG_ITEM_PICKUP, .proc/RemoveItemFromTable) //Listen for the pickup event, unregister on pick-up so we aren't moved

```

Comment:
There are a shitload of more way to remove something from a table other than pickup. Dragging, knockback, deletion, TK. List goes on. Should probably just listen to moved.

Reply:
good idea thank u gamer

Reply:
do you know if theres a moved signal

Reply:
nvm i find